File: Objc/PixelPixie/Node/PPPixieNode.m

#import "PPPixieNode.h"

@implementation PPPixieNode

- (id)initWithPixie:(PPPixie *)pixie {
    if (self = [super init]) {
        self.size = CGSizeMake(BLOCK_WIDTH, BLOCK_HEIGHT);
        self.anchorPoint = CGPointMake(0.0f, 0.0f);
        self.color = [UIColor grayColor];
        
        [self refreshByPixie:pixie];
    }
    return self;
}

// 更新图像
- (void)refreshByPixie:(PPPixie *)pixie {
    self.alpha = 1.0f;
    NSString * imageName = [PPDataUtil getPicByPixie:pixie AndNumber:0];
    self.texture = [SKTexture textureWithImageNamed:imageName];
    
    [self runAction:[PPNodeUtil getActionByPixie:pixie]];
}

- (void)clean {
    self.alpha = 0.0f;
}

@end

File: Swift/PixelPixie/Node/PPPixieNode.swift
//
//  File.swift
//  PixelPixie
//
//  Created by Junchao Yu on 2018/11/17.
//  Copyright © 2018年 ！. All rights reserved.
//

@objc
class PPPixieNode: SKSpriteNode {
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    init(pixie: PPPixie) {
        super.init(texture: nil,
                   color: UIColor.clear,
                   size: CGSize(width: CGFloat(BLOCK_WIDTH), height: CGFloat(BLOCK_HEIGHT)))
        
        self.anchorPoint = CGPoint(x:0, y:0)
        self.refresh(byPixie: pixie)
    }
    
    func refresh(byPixie pixie: PPPixie) -> Void{
        self.alpha = 1.0
        let imageName = PPDataUtil.getPicBy(pixie, andNumber: 0)
        
        self.texture = SKTexture.init(imageNamed: imageName!)
        self.run(PPNodeUtil.getActionBy(pixie))
    }
    
    func clean() -> Void {
        self.alpha = 0.0
    }
}
